src/main/java/uk/gov/justice/digital/delius/transformers/ConvictionTransformer.java

```diff
@@ -0,0 +1,98 @@
+        this.additionalOffenceTransformer = additionalOffenceTransformer;
+    }
+
+    public Conviction convictionOf(Event event) {
+        return Conviction.builder()
+                .active(zeroOneToBoolean(event.getActiveFlag()))
+                .convictionDate(event.getConvictionDate())
+                .referralDate(event.getReferralDate())
+                .convictionId(event.getEventId())
+                .index(indexOf(event.getEventNumber()))
+                .offences(offencesOf(event))
+                .sentence(Optional.ofNullable(event.getDisposal()).map(this::sentenceOf).orElse(null))
+                .inBreach(zeroOneToBoolean(event.getInBreach()))
+                .latestCourtAppearanceOutcome(Optional.ofNullable(event.getCourtAppearances()).map(this::outcomeOf).orElse(null))
+                .build();
+    }
+
+    private Long indexOf(String eventNumber) {
+        // no DB constraint so protected ourselves
+        return Optional.ofNullable(eventNumber)
+                .filter(StringUtils::hasText)
+                .filter(this::isNumber)
+                .map(text -> NumberUtils.parseNumber(text, Long.class))
+                .orElse(null);
+    }
+
+    private boolean isNumber(String text) {
+        return text.matches("\\d*");
+    }
+
+    private KeyValue outcomeOf(List<CourtAppearance> courtAppearances) {
+        return courtAppearances
+                .stream()
+                .filter(courtAppearance -> courtAppearance.getOutcome() != null).max(Comparator.comparing(CourtAppearance::getAppearanceDate))
+                .map(courtAppearance -> outcomeOf(courtAppearance.getOutcome()))
+                .orElse(null);
+    }
+    private KeyValue outcomeOf(StandardReference outcome) {
+        return KeyValue
+                .builder()
+                .description(outcome.getCodeDescription())
+                .code(outcome.getCodeValue())
+                .build();
+    }
+
+
+    private List<Offence> offencesOf(Event event) {
+        return ImmutableList.<Offence>builder()
+                .addAll(Optional.ofNullable(event.getMainOffence()).map(mainOffence -> ImmutableList.of(mainOffenceTransformer.offenceOf(mainOffence))).orElse(ImmutableList.of()) )
+                .addAll(additionalOffenceTransformer.offencesOf(event.getAdditionalOffences()))
+                .build();
+    }
+
+    private Sentence sentenceOf(Disposal disposal) {
+        return Sentence.builder()
+                .defaultLength(disposal.getLength())
+                .effectiveLength(disposal.getEffectiveLength())
+                .lengthInDays(disposal.getLengthInDays())
+                .originalLength(disposal.getEntryLength())
+                .originalLengthUnits(Optional.ofNullable(disposal.getEntryLengthUnits())
+                        .map(StandardReference::getCodeDescription)
+                        .orElse(null))
+                .secondLength(disposal.getLength2())
```

Comment:
Did we not find a better way of describing "second length"? Sure I badgered Nick R about if for an earlier PR

Reply:
No, and unfortunately the Delius BA essentially described it as multi use - some times just the second half of a sentence (3 years in prison and 4 years on licence) and sometimes for suspended period for suspended sentence. In other words multi-use so we can't find a name that describes it better

Reply:
Actually, I really don't like this. Current understanding is that the field in Delius has 2 meanings, one as the tariff length for a life sentence, and the other the suspended sentence period for suspended sentences. I would propose, then, taking `secondLength` and associated fields out of the API response, and replacing them with either `lifeSentenceTariff` or `suspendedSentencePeriod`

Reply:
The problem is we don't know if it is a `suspendedSentencePeriod ` or a `licenceLength` or a `lifeSentenceTariff `. We just don't have enough context for this given we have to model a legacy database. So at best we would have to fraudulently populate all those fields even though length2 for a particular sentence might mean to the probation service only one of those things. I'm not sure we can change this until we have control on the data written and we can model that correctly. Right now it is literally the thing as described 'a second sentence length'

Reply:
What an utter shambles. It's just a skip whose contents are only ever referenced by the front end, that's what we're saying?